base/loading.jl

```diff
@@ -1859,11 +1862,19 @@ For more details regarding code loading, see the manual sections on [modules](@r
 [parallel computing](@ref code-availability).
 """
 function require(into::Module, mod::Symbol)
-    if _require_world_age[] != typemax(UInt)
-        Base.invoke_in_world(_require_world_age[], __require, into, mod)
-    else
-        @invokelatest __require(into, mod)
+    ex = nothing
+    out = try
+        if _require_world_age[] != typemax(UInt)
+            Base.invoke_in_world(_require_world_age[], __require, into, mod)
+        else
+            @invokelatest __require(into, mod)
+        end
+    catch _ex
+        # don't include loading internals in stacktrace unless requested
+        hide_loading_frames[] ? (ex = _ex) : rethrow()
```

Comment:
This has irked me for awhile too, but I don't think this is a good approach. If the user is running their own package code in the `require` call (as expected for a require call, sans possible precompiling), it will also consume their stacktrace entirely, while we only really want to eliminate the frames corresponding here to loading itself.

Reply:
So find chunks of `loading.jl` and `essentials.jl` frames and replace them with something like this?
```
Stacktrace:
 [1] user_func
   @ Package Package.jl:989 [inlined]
 [2-10] [loading internals]
   @ Base ./loading.jl
 [11] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1868
```

Reply:
yeah, or not even listing `[loading internals]` separately, but combine it with `require`. Or maybe something like:
```
Stacktrace:
 [1] user_func
   @ Package Package.jl:989 [inlined]
 [2-11] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1868
   @ Base ./loading.jl:1
   @ Base ./loading.jl:2
   @ Base ./loading.jl:3
   @ Base ./loading.jl:4
[12] top-level scope
   @ ~/.julia/config/startup.jl:18
```

Reply:
A bit tricker to do but more compact, because loading stacktraces often have both these files
```
Stacktrace:
 [1] user_func
   @ Package Package.jl:989 [inlined]
 [2-11] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1868 :1 :2 :3 :4
   @ Base ./essentials.jl:24 :56
   @ Base ./loading.jl:10 :743
[12] top-level scope
   @ ~/.julia/config/startup.jl:18
 ```

But yeah, I think maybe just this would be fine

```
Stacktrace:
 [1] user_func
   @ Package Package.jl:989 [inlined]
[2-11] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1868 (internals hidden)
[12] top-level scope
   @ ~/.julia/config/startup.jl:18
 ```

Reply:
I'm a fan of that last (smallest) example. The "real" link for user code is from the `require()` to the `user_func`, so I like the way that's folded.

The stack-trace internal to loading seems like an implementation detail, so I don't mind it being elided entirely.